component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/BeamProducerFinder.java

```diff
@@ -15,44 +15,47 @@
  */
 package org.talend.sdk.component.runtime.beam.spi;
 
-import java.io.ObjectStreamException;
-import java.io.Serializable;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Queue;
-import java.util.concurrent.ArrayBlockingQueue;
-
+import org.apache.beam.runners.direct.DirectOptions;
+import org.apache.beam.runners.direct.DirectRunner;
 import org.apache.beam.sdk.Pipeline;
 import org.apache.beam.sdk.PipelineResult;
-import org.apache.beam.sdk.options.PipelineOptions;
 import org.apache.beam.sdk.options.PipelineOptionsFactory;
 import org.apache.beam.sdk.transforms.DoFn;
 import org.apache.beam.sdk.transforms.PTransform;
 import org.apache.beam.sdk.transforms.ParDo;
 import org.apache.beam.sdk.values.PBegin;
 import org.apache.beam.sdk.values.PCollection;
+
+import lombok.extern.slf4j.Slf4j;
 import org.talend.sdk.component.api.record.Record;
 import org.talend.sdk.component.api.service.source.ProducerFinder;
 import org.talend.sdk.component.runtime.base.Delegated;
-import org.talend.sdk.component.runtime.base.LifecycleImpl;
 import org.talend.sdk.component.runtime.input.Input;
 import org.talend.sdk.component.runtime.input.Mapper;
 import org.talend.sdk.component.runtime.manager.service.ProducerFinderImpl;
 import org.talend.sdk.component.runtime.manager.service.api.ComponentInstantiator;
 import org.talend.sdk.component.runtime.serialization.SerializableService;
 
-import lombok.extern.slf4j.Slf4j;
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ArrayBlockingQueue;
 
 @Slf4j
 public class BeamProducerFinder extends ProducerFinderImpl {
 
     static final int CAPACITY = Integer.parseInt(System.getProperty("talend.beam.wrapper.capacity", "100000"));
 
-    static final Queue<Record> QUEUE = new ArrayBlockingQueue<>(CAPACITY, true);
+    private static List<Queue<Record>> QUEUE = new ArrayList<>();
```

Comment:
Use a Map instead a List to allow deletion of Queue after using it, as
`private static Map<UUID, Queue<Record>> QUEUEs = new ConcurrentHashMap<>();`
(then, a random generated UUID will replace the index)